app/Repositories/CircleRepository.php

```diff
@@ -113,6 +124,53 @@ public function getWebhook($circle_id)
         return $circle->discord_webhook ?: '';
     }
 
+    public function fullCircleData($request, $circle_id)
+    {
+        $profile = $request->user();
+        $user = $profile->users()->where('circle_id', $circle_id)->first();
+        if ($profile->admin_view || $user) {
+            $nominees = Nominee::where('circle_id', $circle_id)->get();
+            $users = User::where('circle_id', $circle_id)->get();
+            $latestEpoch = Epoch::where('circle_id', $circle_id)->whereNotNull('number')->orderBy('number', 'desc')->first();
+            $latestEpochId = $latestEpoch ? $latestEpoch->id : null;
+            $token_gifts = [];
+            $pending_gifts = [];
+            if ($latestEpoch) {
+                if ($latestEpoch->ended == 1) {
+                    $token_gifts = Utils::queryCache($request, function () use ($circle_id, $user, $latestEpochId) {
```

Comment:
Just to check this is only for the latestEpoch?

Reply:
yeah followed this comment `most recent epoch` 
https://linear.app/coordinape/issue/DEV-282#comment-05781993